### PR TITLE
Adds configuration files

### PR DESCRIPTION
**Pull request by PascalSenn**

### CODE REVIEW

**Review by PascalSenn — COMMENTED** [review]
LGTM

File: src/Confix.Tool/src/Confix.Tool/ConfigurationFile/IConfigurationFileProvider.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+using Confix.Tool.Middlewares.Project;
+using Factory =
```
Comment (RohrerF) [inline src/Confix.Tool/src/Confix.Tool/ConfigurationFile/IConfigurationFileProvider.cs:2]: unused

File: src/Confix.Tool/src/Confix.Tool/Middlewares/Project/BuildProjectMiddleware.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
+using System.Text.Json.Nodes;
+using Confix.Tool.Commands.Logging;
+using Confix.Tool.Common.Pipelines;
+using Confix.Tool.Schema;
+using Spectre.Console;
+
+namespace Confix.Tool.Middlewares.Project;
+
+public sealed class BuildProjectMiddleware : IMiddleware
```
Comment (RohrerF) [inline src/Confix.Tool/src/Confix.Tool/Middlewares/Project/BuildProjectMiddleware.cs:9]: please add test for this

File: src/Confix.Tool/src/Confix.Tool/ConfigurationFile/ConfigurationFile.cs [hunk 3]
```diff
@@ -0,0 +1,64 @@
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json.Nodes;
+using Confix.Tool.Commands.Logging;
+using Confix.Tool.Schema;
+using Microsoft.Extensions.Logging;
+
+namespace Confix.Tool.Middlewares;
+
+/// <summary>
```
Comment (RohrerF) [inline src/Confix.Tool/src/Confix.Tool/ConfigurationFile/ConfigurationFile.cs:9]: doesn't match :)

File: src/Confix.Tool/src/Confix.Tool/ConfigurationFile/ConfigurationFileMiddleware.cs [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    private readonly IConfigurationFileProviderFactory _factory;
+
+    public ConfigurationFileMiddleware(IConfigurationFileProviderFactory factory)
+    {
+        _factory = factory;
+    }
+
+    /// <inheritdoc />
+    public async Task InvokeAsync(IMiddlewareContext context, MiddlewareDelegate next)
+    {
```
Comment (RohrerF) [inline src/Confix.Tool/src/Confix.Tool/ConfigurationFile/ConfigurationFileMiddleware.cs:20]: Test if testable

File: src/Confix.Tool/.idea/.idea.Confix.Tool/.idea/workspace.xml
Comment (RohrerF) [inline src/Confix.Tool/.idea/.idea.Confix.Tool/.idea/workspace.xml:1]: kill your git cache to finally get rid of this file

**Review by RohrerF — APPROVED** [review]

**Pull request merged**